### PR TITLE
*: support variable tidb_last_ddl_info

### LINKED ISSUE #32402: support session variable `tidb_last_ddl_info`
## Feature Request

**Is your feature request related to a problem? Please describe:**

Since TiDB already support `tidb_last_txn_info`, it's reasonable to support `tidb_last_ddl_info`.
The client may want to know the order of the concurrent DDLs. We want to give this order info in `tidb_last_ddl_info`.
For example, we have two DDLs, namely DDL1 and DDL2, sent by two sessions concurrently.
We can't know which DDL is executed first.
After introducing `tidb_last_ddl_info`, we can use `select @@tidb_last_ddl_info` to get `x` for the first executed DDL and `x+1` for the second executed DDL.

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #32402

Problem Summary:

### What is changed and how it works?
support variable tidb_last_ddl_info. It would return the query and ddl sequence for now.
The sequence of a DDL is the count of the DDLs before it.
For the first DDL, it is 0, and the second DDL is 1, and so on.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
support variable tidb_last_ddl_info
```

### CODE REVIEW

**Review by morgo — APPROVED** [review]

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -3845,3 +3845,28 @@ func TestInvalidPartitionNameWhenCreateTable(t *testing.T) {
...
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec(`use test;`)
+	tk.MustQuery("select json_extract(@@tidb_last_ddl_info, '$.query'), json_extract(@@tidb_last_ddl_info, '$.seq_num')").Check(testkit.Rows("\"\" 0"))
+	tk.MustExec("create table t(a int)")
+	firstSequence := 0
+	res := tk.MustQuery("select json_extract(@@tidb_last_ddl_info, '$.query'), json_extract(@@tidb_last_ddl_info, '$.seq_num')")
+	require.Equal(t, 1, len(res.Rows()))
```
Comment (hawkingrei) [inline ddl/db_integration_test.go:3859]: use ```require.Len```.

File: ddl/db_integration_test.go [hunk 2]
```diff
@@ -3845,3 +3845,28 @@ func TestInvalidPartitionNameWhenCreateTable(t *testing.T) {
...
+	require.Equal(t, "\"create table t(a int)\"", res.Rows()[0][0])
+	var err error
+	firstSequence, err = strconv.Atoi(fmt.Sprintf("%v", res.Rows()[0][1]))
+	require.Nil(t, err)
```
Comment (hawkingrei) [inline ddl/db_integration_test.go:3863]: use ```require.NoError```.

**Review by xiongjiwei — APPROVED** [review]

**Comment by hawkingrei** [comment]
Please fix the linter.

**Pull request merged**